samples/nrf_desktop/src/hw_interface/battery_discrete.c

```diff
@@ -389,15 +420,17 @@ static bool event_handler(const struct event_header *eh)
 				module_set_state(MODULE_STATE_READY);
 				atomic_set(&active, true);
 			}
-
-			return false;
+		} else if (event->state == MODULE_STATE_ERROR) {
+			battery_state = BATTERY_STATE_ERROR;
+			sleep();
 		}
 
 		return false;
 	}
 
 	if (is_wake_up_event(eh)) {
-		if (!atomic_get(&active)) {
+		if (!atomic_get(&active) &&
+		    battery_state != BATTERY_STATE_ERROR) {
```

Comment:
I was thinking about it and I don't like the idea of setting error to state if no error was seen in this module. 

There is a way of making it work without doing these things.
In every module on error case perform shutdown action (like you do). Make error module sign up early for wake up event and make it swallow it so nothing else will be able to wake up. This way board will remain shutdown except error module.

Reply:
Discussed offline